esp32_ulp/opcodes.py

```diff
@@ -285,13 +285,38 @@ def eval_arg(arg):
             _, _, sym_value = symbols.get_sym(token)
             parts.append(str(sym_value))
         else:
-            parts.append(token)
+            try:
+                # attempt to parse, to convert numbers with base prefix correctly
+                int_token = parse_int(token)
+                parts.append(str(int_token))
+            except ValueError:
+                parts.append(token)
     parts = "".join(parts)
     if not validate_expression(parts):
         raise ValueError('Unsupported expression: %s' % parts)
     return eval(parts)
 
 
+def parse_int(literal):
```

Comment:
I'm not familiar with this code base, but would it make sense to factor this function out into a separate file, so it can be reused in `opcodes_s2.py`?

Similarly, could have a single unit test for this function in a separate testing file.

(Just a suggestion :smile: )